src/client/models/generateDamage.js

```diff
@@ -0,0 +1,317 @@
+        xPrev = segmentPairPoints[i - 1][0];
+        yPrev = segmentPairPoints[i - 1][1];
+        xCurr = segmentPairPoints[i][0];
+        yCurr = segmentPairPoints[i][1];
+
+        pointsOnDamageLine = findIntersectionCoordinates(xPrev, yPrev, xCurr, yCurr, damageX, damageY, damageRadius);
+        intersectPt1 = pointsOnDamageLine[0];
+        intersectPt2 = pointsOnDamageLine[1];
+        intersectPt1X = intersectPt1[0];
+        intersectPt1Y = intersectPt1[1];
+        intersectPt2X = intersectPt2[0];
+        intersectPt2Y = intersectPt2[1];
+
+        setPointOrder(xPrev, yPrev, xCurr, yCurr, intersectPt1X, intersectPt1Y, intersectPt2X, intersectPt2Y, pointsToReplace);
+    }
+
+    pointsToReplace.push(segmentPairPoints[segmentPairPoints.length - 1]);
+
+    return pointsToReplace;
+};
+
+// helper-functions
+
+const markAndPushPoint = (point, arrayInOrder) => {
+    point.push('inDamage');
+    arrayInOrder.push(point);
+};
+
+const setPointOrder = (endpoint1X, endpoint1Y, endpoint2X, endpoint2Y, damagePoint1X, damagePoint1Y, damagePoint2X, damagePoint2Y, arrayOfOrder) => {
+    // compare and set order of two damagePoint's T's coefficients of two points, situated on the same line segment
+    let damagePoint1T,
+        damagePoint2T,
+        damagePoint1 = [damagePoint1X, damagePoint1Y],
+        damagePoint2 = [damagePoint2X, damagePoint2Y],
+        initialCheck1,
+        initialCheck2,
+        initialCheck3,
+        initialCheck4;
+
+    damagePoint1T = findLineSegmentCoefficient(endpoint1X, endpoint1Y, endpoint2X, endpoint2Y, damagePoint1X, damagePoint1Y);
+    damagePoint2T = findLineSegmentCoefficient(endpoint1X, endpoint1Y, endpoint2X, endpoint2Y, damagePoint2X, damagePoint2Y);
+
+    initialCheck1 = (0 <= damagePoint1T && damagePoint1T <= 1);
+    initialCheck2 = (0 <= damagePoint2T && damagePoint2T <= 1);
+    initialCheck3 = (!initialCheck1 && initialCheck2);
+    initialCheck4 = (initialCheck1 && !initialCheck2);
+
+    if ( (initialCheck1 && initialCheck2) && damagePoint1T < damagePoint2T ) {
+        markAndPushPoint(damagePoint1, arrayOfOrder);
+        markAndPushPoint(damagePoint2, arrayOfOrder);
+    } else if (initialCheck3) {
+        markAndPushPoint(damagePoint2, arrayOfOrder);
+    } else if (initialCheck4) {
+        markAndPushPoint(damagePoint1, arrayOfOrder);
+    } else if ( (initialCheck1 && initialCheck2) && damagePoint1T > damagePoint2T ) {
+        markAndPushPoint(damagePoint2, arrayOfOrder);
+        markAndPushPoint(damagePoint1, arrayOfOrder);
+    }
+};
+
+const findLineSegmentCoefficient = (endpoint1X, endpoint1Y, endpoint2X, endpoint2Y, damagePointX, damagePointY) => {
+    // find coefficient of point, situated on line segment
+    let deltaX,
+        deltaY;
+
+    deltaX = (endpoint2X - endpoint1X);
+    deltaY = (endpoint2Y - endpoint1Y);
+
+    if (deltaX != 0) {
+        return ( (damagePointX - endpoint1X) / deltaX );
+    } else {
```

Comment:
Thanks.
Actually this is a mistake in code logic - already fixing it on my branch